### PR TITLE
Add Firebase Dynamic Links URL Shortener

### LINKED ISSUE #3272: Remove goo.gl
"Starting March 30, 2018, we will be turning down support for goo.gl URL shortener. From April 13, 2018 only existing users will be able to create short links on the goo.gl console. You will be able to view your analytics data and download your short link information in csv format for up to one year, until March 30, 2019, when we will discontinue goo.gl. Previously created links will continue to redirect to their intended destination. Please see this [blog post](https://developers.googleblog.com/2018/03/transitioning-google-url-shortener.html) for more details"

https://goo.gl/
https://developers.googleblog.com/2018/03/transitioning-google-url-shortener.html

So i think it would be better to remove it sooner then later so less people start using it then get annoyed and/or have to redo their ShareX setup with another url shorten service

### PR DESCRIPTION
**Pull request by matthewburnett**
Fixes #3272

### CODE REVIEW

File: ShareX.UploadersLib/URLShorteners/FirebaseDynamicLinksURLShortener.cs [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+                FirebaseWebAPIKey = config.FirebaseWebAPIKey,
+                DynamicLinkDomain = config.FirebaseDynamicLinkDomain,
+                IsShort = config.FirebaseIsShort
+            };
+        }
+
+        public override TabPage GetUploadersConfigTabPage(UploadersConfigForm form) => form.tpFirebaseDynamicLinks;
+    }
+
+    public class FirebaseResponse
```
Comment (sylveon) [inline ShareX.UploadersLib/URLShorteners/FirebaseDynamicLinksURLShortener.cs]: Even if you don't need them right now, add all of the response's fields

File: ShareX.UploadersLib/Forms/UploadersConfigForm.resx [hunk 2]
```diff
@@ -11775,7 +11925,7 @@ Using an encrypted library disables sharing.</value>
     <value>4, 4, 4, 4</value>
   </data>
   <data name="ucLocalhostAccounts.Size" type="System.Drawing.Size, System.Drawing">
-    <value>792, 999</value>
+    <value>792, 2037</value>
```
Comment (sylveon) [inline ShareX.UploadersLib/Forms/UploadersConfigForm.resx]: lol?

Comment (matthewburnett) [reply ShareX.UploadersLib/Forms/UploadersConfigForm.resx]: What's wrong with that?

Comment (sylveon) [reply ShareX.UploadersLib/Forms/UploadersConfigForm.resx]: Why does the control randomly doubles in height

**Pull request merged**